src/managers/PointerManager.cpp

```diff
@@ -676,8 +678,11 @@ void CPointerManager::warpTo(const Vector2D& logical) {
     damageIfSoftware();
 
     pointerPos = closestValid(logical);
-    recheckEnteredOutputs();
-    onCursorMoved();
+
+    if (!g_pInputManager->isLocked()) {
```

Comment:
what if we are changing a workspace? Wouldn't that lock our cursor to the current window? Why this change?

Reply:
Fast movements can escape the lock briefly and will trigger active monitor switch causing unnecessary frames and frame rate spikes https://github.com/hyprwm/Hyprland/pull/6877#issuecomment-2230423256

Reply:
Changing a workspace doesn't seem to have any issues with this

Reply:
ok